### ISSUE TITLE
consumer.apply(JwtMiddleware).forRoutes(AppController) with setGlobalPrefix dont work 

### ISSUE THREAD

**Issue by igorivaniuk** [nestjs/nest#718]
## I'm submitting a...

<pre><code>
[x] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

when set `app.setGlobalPrefix('api')`

```ts
@Module({
  imports: [],
  controllers: [AppController],
  providers: [ AppService ]
})
export class AppModule {
  configure(consumer: MiddlewareConsumer) {
    consumer.apply(JwtMiddleware).forRoutes(AppController); // not work
    consumer.apply(FooMiddleware).forRoutes('/home'); // not work
    consumer.apply(BarMiddleware).forRoutes('/api/home'); // work
  }
}
```
app.controller.ts
```ts
@Controller('home')
export class AppController {
  constructor() {}
  @Get()
  root(@Req() req: any): any {
    return {
      foo: req.foo,
      bar: req.bar,
      jwt: req.jwt,
    };
  }
}
```

## Expected behavior

```ts
consumer.apply(JwtMiddleware).forRoutes(AppController); // work
consumer.apply(FooMiddleware).forRoutes('/home'); // work
consumer.apply(BarMiddleware).forRoutes('/api/home'); // work
```

## Minimal reproduction of the problem with instructions

https://github.com/igorivaniuk/nest-prefix-bug

## Environment

<pre><code>
Nest version: v8.9.4

Nest: 5.0.0 

For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Thanks for reporting!

**Comment by kamilmysliwiec** [comment]
Fixed in `v5.0.1`
